### PR TITLE
add linters config file and Gruntfile.js

### PR DESCRIPTION
**Pull request by neerajsachan**
we have split Gruntfile.js into multiple task using grunt plugin.

### CODE REVIEW

File: generators/linter/templates/grunt/eslint.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (gaurav21r) [inline generators/linter/templates/grunt/eslint.js]: Why this space. No spaces!

File: generators/linter/templates/grunt/htmlhint.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+  module.exports = {
+    all: {
+      options: {
+        'tag-pair': true,
```
Comment (gaurav21r) [inline generators/linter/templates/grunt/htmlhint.js]: Why is this here? This is a rule. Should be in htnlhintrc not here!

**Review by gaurav21r — CHANGES_REQUESTED** [review]
1. Why is there no `target` specified in eslint but targets are specified in other tasks?
2. Please make sure eslint checks for ALL js files in the repo except `node_modules` and `bower_components` using `/**/*.js` . Similarly for all linters!

File: generators/linter/templates/grunt/mochaTest.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  test: {
```
Comment (gaurav21r) [inline generators/linter/templates/grunt/mochaTest.js]: @neerajsachan Do we need a `target` here? @riteshe63 Can you check if this is compatible with the mocha.opts file?
@neerajsachan the path is not right! We need to run tests based on the folder structure. So it can be `./test/unit/**/*.js` or something else. @riteshe63 Please advice!

Comment (gaurav21r) [inline generators/linter/templates/grunt/mochaTest.js]: @neerajsachan Do we need a `target` here? @riteshe63 Can you check if this is compatible with the mocha.opts file?
@neerajsachan the path is not right! We need to run tests based on the folder structure. So it can be `./test/unit/**/*.js` or something else. @riteshe63 Please advice!

**Comment by neerajsachan** [comment]
I have change the configuration file of the stylelint.js , eslint.js and add the .lintignore file in generators/linter , add the html plugin in generators/linter, .eslintrc file and also add the `"leslint-plugin-html"` in the index.js file.

**Comment by neerajsachan** [comment]
Delete the Gruntfile.js from the generators/linter/templates/ , folder and also change the configuration file  tslint.js ./.linterignore instead of ./.lintignore .

**Comment by neerajsachan** [comment]
change the tslint.js file ./.linterignore instead of  ./.lintignore .

**Comment by neerajsachan** [comment]
Create  .eslintrc file using mocha framework. #54

**Comment by gaurav21r** [comment]
@neerajsachan Please do:
1. Update `devDependencies` to include **minimum** dependencies we need after converting linters to grunt plugins.
2. Use yeoman's `copyTpl` to copy gruntfiles.

**Comment by neerajsachan** [comment]
@gaurav21r after installing grunt plugins we need the plugins `grunt-eslint` for JavaScript file , `grunt-stylelint` and `stylelint-config-standard` for the .css file , `grunt-htmlhint` for the .html file and the grunt-tslint and `tslint-microsoft-contrib` for the typescript file . But eslint plugin,i.e.,`eslint-plugin-html` is not included  in .html file, although we are still working on it.

**Comment by riteshe63** [comment]
@neerajsachan .eslintrc is in use for `airbnb` module and this module is not define, So please check again.

**Comment by neerajsachan** [comment]
@riteshe63 `airbnb` module use in .eslintrc but when install the `grunt-eslint` , `airbnb` module install automatically so we don't  need to install `airbnb` module specially.

**Comment by riteshe63** [comment]
Please check.

**Comment by riteshe63** [comment]
@gaurav21r Add task ruuner to linter-generator.

File: generators/linter/templates/grunt/aliases.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (gaurav21r) [inline generators/linter/templates/grunt/aliases.js]: @riteshe63 We shouldn't have a file for aliases. We are changing it via code inside `index.js`. Having a file will not allow us to use `merge`

File: generators/linter/index.js [hunk 5]
```diff
@@ -48,25 +48,38 @@ module.exports = yeoman.Base.extend({
...
     // Read a package.json if it exists or create it
     const packageJSON = this.fs.readJSON(this.destinationPath('package.json'), {});
+    let aliasesJS = require(this.destinationPath('./grunt/aliases.js'));
     let atomPackages = _includes(this.selectedEditors, 'Atom') ?
       this.fs.read(this.destinationPath('atom-packages.txt'), { defaults: '' }).split('\n') : null;
 
     // JAVASCRIPT
     if (_includes(this.selectedLanguages, 'JavaScript')) {
       this.log('Configuring ESLint for Linting JavaScript');
+      aliasesJS = 'module.exports=' + JSON.stringify(_merge(aliasesJS, {
```
Comment (gaurav21r) [inline generators/linter/index.js]: @riteshe63 This is inside the selection of JavaScript so we shouldn't be including stylelint here. What if someone doesn't select CSS? This will still execute. 

@riteshe63 Please try executing this code and lemme know if it works

``` js
aliasesJS.lint = aliasesJS.lint || []
aliasesJS.lint.push('eslint')
```

Similarly for CSS,

``` js
aliasesJS.lint = aliasesJS.lint || []
aliasesJS.lint.push('stylelint')
```

File: generators/linter/index.js [hunk 6]
```diff
@@ -126,6 +143,7 @@ module.exports = yeoman.Base.extend({
 
     this.log('Writing to package.json');
     this.fs.writeJSON(this.destinationPath('package.json'), packageJSON);
+    this.fs.write(this.destinationPath('grunt/aliases.js'), aliasesJS);
 
     // editorconfig
     this.log('Writing .editorconfig');
```
Comment (gaurav21r) [inline generators/linter/index.js:160]: @riteshe63 There is no `editorconfig` in `/templates` :cry:

**Review by gaurav21r — CHANGES_REQUESTED** [review]

File: generators/linter/templates/grunt/aliases.js (hunk 4 above)
Comment (riteshe63) [reply generators/linter/templates/grunt/aliases.js]: Done

File: generators/linter/index.js (hunk 5 above)
Comment (riteshe63) [reply generators/linter/index.js]: Done

File: generators/linter/index.js (hunk 6 above)
Comment (riteshe63) [reply generators/linter/index.js:160]: Done

**Comment by riteshe63** [comment]
All done
